src/Maths/Silk.NET.Maths/Scalar.cs

```diff
@@ -0,0 +1,845 @@
+            }
+        }
+
+        public static T PositiveInfinity
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            get
+            {
+                if (typeof(T) == typeof(Half))
+                {
+                    return (T)(object)Half.PositiveInfinity;
+                }
+
+                if (typeof(T) == typeof(float))
+                {
+                    return (T)(object)float.PositiveInfinity;
+                }
+
+                if (typeof(T) == typeof(double))
+                {
+                    return (T)(object)double.PositiveInfinity;
+                }
+
+                ThrowInvalidType();
+                return default; // can't be reached
+            }
+        }
+
+        public static T NegativeInfinity
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            get
+            {
+                if (typeof(T) == typeof(Half))
+                {
+                    return (T)(object)Half.NegativeInfinity;
+                }
+
+                if (typeof(T) == typeof(float))
+                {
+                    return (T)(object)float.NegativeInfinity;
+                }
+
+                if (typeof(T) == typeof(double))
+                {
+                    return (T)(object)double.NegativeInfinity;
+                }
+
+                ThrowInvalidType();
+                return default; // can't be reached
+            }
+        }
+        
+        internal static void ThrowInvalidType()
+        {
+            throw new NotSupportedException(); // TODO: Document what went wrong
+        }
+
+        internal static void ThrowForUnsupportedBaseType()
+        {
+            if (
+                typeof(T) != typeof(byte) &&
+                typeof(T) != typeof(sbyte) &&
+                typeof(T) != typeof(ushort) &&
+                typeof(T) != typeof(short) &&
+                typeof(T) != typeof(uint) &&
+                typeof(T) != typeof(int) &&
+                typeof(T) != typeof(ulong) &&
+                typeof(T) != typeof(long) &&
+                typeof(T) != typeof(Half) &&
+                typeof(T) != typeof(float) &&
+                typeof(T) != typeof(double))
+                ThrowInvalidType();
+        }
+
+        internal static void ThrowHalfNotSupported()
+        {
+            throw new NotSupportedException("Half not supported by the underlying type");
+        }
+
+        internal static void ThrowIndexOutOfRange()
+        {
+            throw new IndexOutOfRangeException();
+        }
+        
+        internal static void ThrowVectorTTooSmall()
+        {
+            throw new NotSupportedException("Vector<T> too small too fit");
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static T SquareRoot(T value)
+        {
+            if (typeof(T) == typeof(byte))
+            {
+                return (T) (object) (MathF.Sqrt((byte) (object) value));
```

Comment:
MathF isn't available in .NET Standard 2.0, only 2.1 at the earliest. Is it possible to make do with the regular `Math` type?

Reply:
Theoretically yes, but I'd think it can affect performance.
I've added a `using MathF = System.Math` when .Net Standard 2.0 is used.